File: core/src/main/java/heroes/journey/tilemap/wavefunctiontiles/Terrain.java
package heroes.journey.tilemap.wavefunctiontiles;

import static heroes.journey.registries.Registries.TerrainManager;

import org.jetbrains.annotations.NotNull;

import heroes.journey.registries.Registrable;
import lombok.Getter;

@Getter
public class Terrain extends Registrable {

    private final int terrainCost;

    public Terrain(String id, String name, int terrainCost) {
        super(id, name);
        this.terrainCost = terrainCost;
    }

    @NotNull
    @Override
    public Registrable register() {
        return TerrainManager.register(this);
    }
}

File: core/src/main/java/heroes/journey/tilemap/wavefunctiontiles/Terrain.kt
package heroes.journey.tilemap.wavefunctiontiles

import heroes.journey.modlib.registries.Registrable
import heroes.journey.modlib.worldgen.ITerrain
import heroes.journey.mods.Registries

class Terrain(id: String, override val terrainCost: Int) : Registrable(id), ITerrain {
    override fun register(): Terrain {
        return Registries.TerrainManager.register(this)
    }
}
